test/functional/wallet_pruning.py

```diff
@@ -0,0 +1,148 @@
+
+        # Add mature coins to the wallet
+        self.generatetoaddress(self.nodes[0], 1, self.nodes[0].getnewaddress(), sync_fun=lambda: None)
+        self.generate(self.nodes[0], 100)
+
+        # Import wallet into node 1
+        self.nodes[1].createwallet(wallet_name="wallet_pruned", descriptors=self.options.descriptors, load_on_startup=True)
+        self.nodes[1].importwallet(os.path.join(self.nodes[0].datadir, wname))
+
+        # Make sure the wallets have the same balance
+        assert_equal(self.nodes[1].getbalance(), self.nodes[0].getbalance())
+
+        # Mine some blocks, pruning should not have removed the blocks required for the wallet
+        self.mine_large_blocks(self.nodes[0], 5)
+        self.sync_all()
+
+        # Import wallet, should still work
+        self.nodes[1].importwallet(os.path.join(self.nodes[0].datadir, wname))
+        self.log.info("Wallet successfully imported on pruned node")
+
+    def test_wallet_import_pruned_with_missing_blocks(self):
+        self.log.info("Make sure we cannot import wallet when pruned and required blocks are not available")
+
+        wname = "wallet_init.dat"
+
+        # Get birthheight of wallet
+        with open(os.path.join(self.nodes[0].datadir, wname), 'r', encoding="utf8") as f:
+            for line in f:
+                if line.startswith('# * Best block at time of backup'):
+                    wallet_birthheight = int(line.split(' ')[9])
+                    break
+
+        assert_raises_rpc_error(-1, "Block not available (pruned data)", self.nodes[1].getblock, self.nodes[1].getblockhash(wallet_birthheight))
+
+        # Make sure wallet cannot be imported because of missing blocks
+        # This will try to rescan blocks `TIMESTAMP_WINDOW` (2h) before the wallet birthheight.
+        # There are 6 blocks an hour, so 11 blocks (excluding birthheight).
+        assert_raises_rpc_error(-4, f"Pruned blocks from height {wallet_birthheight - 11} required to import keys. Use RPC call getblockchaininfo to determine your pruned height.", self.nodes[1].importwallet, os.path.join(self.nodes[0].datadir, wname))
+
+    def run_test(self):
+        self.log.info("Warning! This test requires ~1.2GB of disk space")
+
+        self.log.info("Generate a long chain of blocks...")
+
+        # A blk*.dat file is 128MB
+        # Generate 288 light blocks, the minimum required to stay on disk with prune enabled
+        self.generate(self.nodes[0], 288, sync_fun=lambda: None)
+        # Also generate 128MB worth of large blocks to fill the first blk*.dat file because it cannot be pruned
+        self.mine_large_blocks(self.nodes[0], 128)
+
+        # Generate 50MB worth of large blocks in the second blk*.dat file
+        self.mine_large_blocks(self.nodes[0], 50)
+        # Create a wallet and a backup of it which birth's block is in the second blk*.dat file
+        self.nodes[0].createwallet(wallet_name="wallet_init", descriptors=self.options.descriptors, load_on_startup=True)
```

Comment:
In 6abd81b11dd1a45bdffcc85ba90591474a3f48b0 "test: Wallet imports on pruned nodes"

Since this is testing legacy wallet only RPCs, we should not be looking at the `--legacy-wallet`/`--descriptors` options to determine the wallet type to create, for all wallets created in this test.

```suggestion
        self.nodes[0].createwallet(wallet_name="wallet_init", descriptors=False, load_on_startup=True)
```